### PR TITLE
Grape Plant Disease Recognition

### PR DESCRIPTION
**Pull request by aditya0929**
# Pull Request for DL-Simplified 💡

## Issue Title : Grape Plant Disease Recognition #342 

- **Info about the related issue (Aim of the project)** : to correctly identify the disease with the help of grape plant leave images

- **Name:** Aditya Narayan Jha
- **GitHub ID:**  https://github.com/aditya0929
- **Email ID:**   adityajha8906@gmail.com
- communication. -->
- **Idenitfy yourself: (Mention in which program you are contributing in. Eg. For a JWOC 2022 participant it's, `JWOC Participant`)** `ssoc-2023`

Closes: #342 

### Describe the add-ons or changes you've made 📃

I chose convolutional neural networks for the multi instance classification situation since the dataset included contained 4 different classes of grape plant diseases and every class had to be taken care off so that the overall accuracy increases of the model .

 `resnet50v2` , `vgg19` and `mobilenet` were the pretrained models that i had chosen for the classification of plant disease of grapes .

`ResNet-50v2` It's an improved version of ResNet-50, using residual connections to address the vanishing gradient problem. It has 50 layers and performs well in image classification tasks.

`MobileNet` It's designed for mobile and embedded devices with limited resources. MobileNet reduces model size and complexity using depth-wise separable convolutions. It's efficient and suitable for real-time applications.

Both ResNet-50v2 and MobileNet have their strengths and are suitable for different scenarios depending on the available resources and the desired trade-offs between accuracy and computational efficiency.

`VGG16`is a powerful deep convolutional neural network with 16 layers, renowned for its strong performance in image recognition tasks and serving as a benchmark model in computer vision research.

I have also included the learning tools that were very helpful to me in developing the model and that include a lot more in-depth information.

**Accuracy Comparison**

| `Model`  | `Accuracy` |
|--------|----------|
| VGG16  |   88%    |
| MobileNet | 97% |
| ResNet50V2 | 93% |
## Type of change ☑️

What sort of change have you made:

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Code style update (formatting, local variables)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested? ⚙️
 

To keep a clean code and concentrate on one model at a time, I decided to take the approach of creating separate ipynb files for each models. Aside from the model architecture, all models' code adheres to the same process for all other aspects.

I tested it by displaying the labels of the original images alongside the predicted labels by the model to check whether both of them are same or not and majority of the time due to the high level of accuracy the test labels and predicted labels came to be the same .

## Checklist: ☑️

- [x] My code follows the guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly wherever it was hard to understand.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added things that prove my fix is effective or that my feature works.
- [x] Any dependent changes have been merged and published in downstream modules.

predicted labels 
image

### CODE REVIEW

**Review by abhisheks008 — APPROVED** [review]
Your PR is approved and ready to be merged.
@aditya0929
